File: backend/models/project.py
from typing import Optional

from pydantic import BaseModel, Field


class ProjectSchema(BaseModel):
    name: str = Field(...)
    description: str = Field(...)
    content: str = Field(...)
    tags: list = Field([])
    archived: bool = Field(False)

    class Config:
        schema_extra = {
            "example": {
                "name": "Default",
                "description": "Default project used to track testing",
                "content": "This can be used to store various *markdown* describing the project.",
                "tags": ["acceptance", "integration", "load", "system"],
                "archived": False
            }
        }


class UpdateProjectModel(BaseModel):
    name: Optional[str]
    description: Optional[str]
    content: Optional[str]
    tags: Optional[list]
    archived: Optional[bool]

    class Config:
        schema_extra = {
            "example": {
                "name": "Default",
                "description": "Default project used to track testing",
                "content": "This can be used to store various *markdown* describing the project.",
                "tags": ["acceptance", "integration", "load", "system"],
                "archived": False
            }
        }

File: backend/src/models/project.ts
import mongoose from 'mongoose'

const dataSchema = new mongoose.Schema({
    name: {
        required: true,
        type: String
    },
    description: {
        required: false,
        type: String
    },
    content: {
        required: false,
        type: String
    },
    versions: {
        required: true,
        type: [String]
    },
    tags: {
        requried: true,
        type: [String]
    }
})

module.exports = mongoose.model('projects', dataSchema)